optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/pages/spotroster/ShiftBlobPopoverContent.java

```diff
@@ -0,0 +1,217 @@
+import org.jboss.errai.ui.shared.api.annotations.Templated;
+import org.optaplanner.openshift.employeerostering.gwtui.client.rostergrid.powers.BlobPopover;
+import org.optaplanner.openshift.employeerostering.gwtui.client.rostergrid.powers.BlobPopoverContent;
+import org.optaplanner.openshift.employeerostering.gwtui.client.rostergrid.view.BlobView;
+import org.optaplanner.openshift.employeerostering.shared.employee.Employee;
+import org.optaplanner.openshift.employeerostering.shared.employee.EmployeeRestServiceBuilder;
+import org.optaplanner.openshift.employeerostering.shared.shift.Shift;
+import org.optaplanner.openshift.employeerostering.shared.shift.ShiftRestServiceBuilder;
+import org.optaplanner.openshift.employeerostering.shared.shift.view.ShiftView;
+import org.optaplanner.openshift.employeerostering.shared.spot.SpotRestServiceBuilder;
+
+import static java.lang.Long.parseLong;
+import static java.util.function.Function.identity;
+import static java.util.stream.Collectors.toMap;
+import static org.optaplanner.openshift.employeerostering.gwtui.client.common.FailureShownRestCallback.onSuccess;
+
+@Templated
+public class ShiftBlobPopoverContent implements BlobPopoverContent {
+
+    @Inject
+    @DataField("root")
+    private HTMLDivElement root;
+
+    @Inject
+    @DataField("close-button")
+    private HTMLButtonElement closeButton;
+
+    @Inject
+    @DataField("from-day")
+    private HTMLInputElement fromDay;
+
+    @Inject
+    @DataField("from-hour")
+    private HTMLInputElement fromHour;
+
+    @Inject
+    @DataField("to-day")
+    private HTMLInputElement toDay;
+
+    @Inject
+    @DataField("to-hour")
+    private HTMLInputElement toHour;
+
+    @Inject
+    @DataField("spot")
+    private ListBox spotSelect; //FIXME: Don't use GWT widget
+
+    @Inject
+    @DataField("employee")
+    private ListBox employeeSelect; //FIXME: Don't use GWT widget
+
+    @Inject
+    @DataField("pinned")
+    private HTMLInputElement pinned;
+
+    @Inject
+    @Named("p")
+    @DataField("rotation-employee")
+    private HTMLElement rotationEmployee;
+
+    @Inject
+    @DataField("delete-button")
+    private HTMLButtonElement deleteButton;
+
+    @Inject
+    @DataField("cancel-button")
+    private HTMLButtonElement cancelButton;
+
+    @Inject
+    @DataField("apply-button")
+    private HTMLButtonElement applyButton;
+
+    private BlobPopover popover;
+
+    private ShiftBlobView blobView;
+
+    private Map<Long, Employee> employeesById;
+
+    @Override
+    public void init(final BlobView<?, ?> blobView) {
+
+        this.blobView = (ShiftBlobView) blobView;
+        final ShiftBlob blob = (ShiftBlob) blobView.getBlob();
+        final Shift shift = blob.getShift();
+
+        employeeSelect.clear();
+        employeeSelect.addItem("Unassigned", "-1"); //FIXME: i18n
+
+        //FIXME: Loading the entire list is not a good idea
+        SpotRestServiceBuilder.getSpotList(shift.getTenantId(), onSuccess(spots -> {
```

Comment:
Should we have the spot list and employee list inside a singleton we can access? I did this with the Skill list

Reply:
Idk, if we have thousands of Employees/Spots this will not be good anyways.. I think we should have a "smart" input that filters the top 10 Employees based on the typed String. Wdyt?

Reply:
We have a JIRA for that, but do we keep the entire list on the client and do the query client-side, or do we do a request to a server for it... (or do it in a Select, which implements this feature (see Employee screen skill field))

Reply:
I think we should consider the use case where we simply can't store the entire employees/spots lists on the client because it's too big, so we should ask the server for a reduced version of the list based on the user input. It's not so simple, that's why I left a FIXME there :P